include/simdjson/generic/ondemand/json_iterator-inl.h

```diff
@@ -28,6 +28,14 @@ simdjson_really_inline json_iterator::json_iterator(const uint8_t *buf, ondemand
   logger::log_headers();
 }
 
+
+inline void json_iterator::rewind() noexcept {
+   token.index = parser->implementation->structural_indexes.get();
+   logger::log_headers(); // We start again
+   _string_buf_loc = parser->string_buf.get();
```

Comment:
Hmm. My intuition is users will be surprised that rewind invalidates unescaped string_views (and confused at the bugs they get). As long as the "you may not unescape a string more than once" rule is followed by the user, they should be fine, and count_elements in particular doesn't unescape. I'd warn here, and *maybe* have a reset_strings method or a boolean parameter to reset strings.

Reply:
Turns out that this was dead code (unused).

Reply:
I was wrong. It was important code.